src/openforms/emails/digest.py

```diff
@@ -69,6 +77,76 @@ class BrokenConfiguration:
     exception_message: str
 
 
+@dataclass
+class InvalidRegistrationBackend:
+    config_name: StrOrPromise
+    exception_message: str
+    form_name: str
+    form_id: str = ""
+    form_level: bool = False
+
+    @property
+    def admin_link(self) -> str:
+        form_admin_url = reverse(
+            "admin:forms_form_change", kwargs={"object_id": self.form_id}
+        )
+        return form_admin_url
+
+
+@dataclass
+class InvalidLogicRule:
+    variable: str
+    form_name: str
+    form_id: str
+
+    @property
+    def admin_link(self) -> str:
+        return reverse("admin:forms_form_change", kwargs={"object_id": self.form_id})
+
+
+def is_form_config_valid(plugin: BasePlugin, options: dict[str, str]) -> bool:
+    serializer = plugin.configuration_options(data=options)
+
+    try:
+        serializer.is_valid(raise_exception=True)
+    except serializers.ValidationError:
+        return False
+
+    return True
+
+
+def check_general_config(plugin: BasePlugin) -> str:
+    try:
+        plugin.check_config()
+
+    # we always raise an InvalidPluginConfiguration exception even when we have
+    # errors like HTTPError
+    except InvalidPluginConfiguration as e:
+        return e.args[0]
+
+    return ""
+
+
+def get_variables_from_logic(form_logic: FormLogic) -> list[str]:
+    data = form_logic.json_logic_trigger
+    var_values = []
+
+    def _recursive_find(data):
```

Comment:
As we discussed, this will need to take into account some edge cases:

<details>

<summary>cases</summary>

```jsonl
{
    "var": "path.to.some.data"
}
{
    "var": ["path.to.some.data", 9]
}
{
    "var": {
        "var": "a"
    }
}
```
</details>

(adding them here so that others can see)

Reply:
also `{"var": ""}` is valid, it returns the whole object